bot/exts/evergreen/quack.py

```diff
@@ -0,0 +1,65 @@
+from typing import Optional
+
+import discord
+from discord.ext import commands
+
+from bot.bot import Bot
+from bot.constants import Colours
+
+
+class Quack(commands.Cog):
+    """Cog for the .quack command."""
+
+    def __init__(self, bot: Bot) -> None:
+        self.bot = bot
+
+    @commands.command()
+    async def quack(self, ctx: commands.Context, seed: Optional[str], ducktype: Optional[str] = None) -> None:
+        """
+        Pulls a random URL from the Quackstack API.
+
+        If a seed is provided, a duck is generated based on the given seed.
+        If a duck type is provided, either a duck or manduck will be generated.
+        """
+        quackstack_url = "https://quackstack.pythondiscord.com/"
+
+        image_url = "https://quackstack.pythondiscord.com"
+
+        if not ducktype:
+            quackstack_url += "duck"
+        else:
+            if ducktype.lower() not in ["duck", "manduck"]:
+                raise commands.BadArgument(
+                    "The given duck type is invalid. The duck type can either be `duck` or `manduck`."
+                )
+            quackstack_url += ducktype.lower()
+        if seed:
+            try:
+                seed = int(seed)
+            except ValueError:
+                newseed = ""
+                for e in seed:
+                    newseed += str(ord(e))
+                seed = int(newseed)
+            quackstack_url += f"?seed={seed}"
+
+        async with self.bot.http_session.get(quackstack_url) as r:
```

Comment:
I think using the `params` argument of `.get` would be preferred here to pass in the seed. You could do something like `params={}` and then in the `if seed` set `params["seed"] = seed` so it is only defined if you use it. Haven't tried that but seems like it should work